### DISCUSSION TITLE
What's the easiest way to get validation loss from SupervisedEvaluator?

### DISCUSSION THREAD

**Discussion by rybarix** [Project-MONAI/MONAI#3786] [Q&A] [answered]
👋 Hello everybody!

I'm wondering what's the clean way to obtain validation loss from supervised evaluator.

It seems there is no loss stored in engine.state.output in SupervisedEvaluator.

**Comment by Nic-Ma** [comment]
Hi @SandroRybarik ,

I think we usually don't compute loss during evaluation, same as ignite:
https://github.com/pytorch/ignite/blob/master/ignite/engine/__init__.py#L560
Or maybe I misunderstood your question?
Please feel free to correct me.

Thanks.

**Comment by rybarix** [comment, marked as answer]
Thank you for quick response.

What I'm trying to achieve is calculate training loss and validation loss during training phase.
I see now that SupervisedEvaluator is not suited for this kind of task.
Is there any other way to get validation loss without rewriting training loop back to plain pytorch?

**Reply by Nic-Ma** [reply]
Hi @SandroRybarik ,

I think you can try this metric in `SupervisedEvalautor` with your loss function:
https://github.com/pytorch/ignite/blob/master/ignite/metrics/loss.py#L11

Thanks.

**Reply by rybarix** [reply]
Yes, this is what I was looking for.

Thank you @Nic-Ma for your guidance.

**Reply by Nic-Ma** [reply]
No problem.

Thanks.
